src/api/java/com/minecolonies/api/colony/colonyEvents/registry/ColonyEventDescriptionTypeRegistryEntry.java

```diff
@@ -0,0 +1,65 @@
+package com.minecolonies.api.colony.colonyEvents.registry;
+
+import com.minecolonies.api.colony.colonyEvents.descriptions.IColonyEventDescription;
+import com.minecolonies.api.util.Log;
+import net.minecraft.nbt.CompoundNBT;
+import net.minecraft.network.PacketBuffer;
+import net.minecraft.util.ResourceLocation;
+import net.minecraftforge.registries.ForgeRegistryEntry;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.function.Function;
+
+/**
+ * This is the colonies event registry entry class, used for registering any colony related events. Takes a function of colony, nbt to create the right event object.
+ */
+public class ColonyEventDescriptionTypeRegistryEntry extends ForgeRegistryEntry<ColonyEventDescriptionTypeRegistryEntry>
```

Comment:
Rework this class so that it does not expose the inner function callbacks.

Reply:
This is how its handled for `IColonyEvent`s tho, does it really have to change?

Reply:
Yes.
The functions are inner information that is not supposed to be exposed to the external world.

Please fix this in IColonyEvent as well.

Reply:
done.